src/seraphis_impl/serialization_demo_utils.cpp

```diff
@@ -419,6 +422,227 @@ void make_serializable_sp_destination_v1(const jamtis::JamtisDestinationV1 &dest
+    origin_context_out.block_timestamp    = origin_context.block_timestamp;
+    origin_context_out.transaction_id     = origin_context.transaction_id;
+    origin_context_out.enote_tx_index     = origin_context.enote_tx_index;
+    origin_context_out.enote_ledger_index = origin_context.enote_ledger_index;
+    origin_context_out.origin_status      = static_cast<unsigned char>(origin_context.origin_status);
+    origin_context_out.tx_extra           = origin_context.memo;
+}
+//-------------------------------------------------------------------------------------------------------------------
+void make_serializable_sp_enote_spent_context_v1(const SpEnoteSpentContextV1 &spent_context, ser_SpEnoteSpentContextV1 &spent_context_out)
+{
+    spent_context_out.block_index     = spent_context.block_index;
+    spent_context_out.block_timestamp = spent_context.block_timestamp;
+    spent_context_out.transaction_id  = spent_context.transaction_id;
+    spent_context_out.spent_status    = static_cast<unsigned char>(spent_context.spent_status);
+}
+//-------------------------------------------------------------------------------------------------------------------
+void make_serializable_sp_enote_variant(const SpEnoteVariant &enote_variant, ser_SpEnoteVariant &enote_variant_out)
+{
+    if (enote_variant.is_type<SpCoinbaseEnoteV1>())
+    {
+        ser_SpCoinbaseEnoteV1 ser_enote;
+        SpCoinbaseEnoteV1 enote = enote_variant.unwrap<SpCoinbaseEnoteV1>();
+        make_serializable_sp_coinbase_enote_v1(enote, ser_enote);
+        enote_variant_out = ser_enote;
+    }
+    else if (enote_variant.is_type<SpEnoteV1>())
+    {
+        ser_SpEnoteV1 ser_enote;
+        SpEnoteV1 enote = enote_variant.unwrap<SpEnoteV1>();
+        make_serializable_sp_enote_v1(enote, ser_enote);
+        enote_variant_out = ser_enote;
+    }
+}
+//-------------------------------------------------------------------------------------------------------------------
+void make_serializable_sp_enote_record_v1(const SpEnoteRecordV1 &enote_record, ser_SpEnoteRecordV1 &enote_record_out)
+{
+    make_serializable_sp_enote_variant(enote_record.enote, enote_record_out.enote);
+    enote_record_out.enote_ephemeral_pubkey = enote_record.enote_ephemeral_pubkey;
+    enote_record_out.input_context          = enote_record.input_context;
+    enote_record_out.enote_view_extension_g = enote_record.enote_view_extension_g;
+    enote_record_out.enote_view_extension_x = enote_record.enote_view_extension_x;
+    enote_record_out.enote_view_extension_u = enote_record.enote_view_extension_u;
+    enote_record_out.amount                 = enote_record.amount;
+    enote_record_out.amount_blinding_factor = enote_record.amount_blinding_factor;
+    enote_record_out.key_image              = enote_record.key_image;
+    memcpy(enote_record_out.address_index.bytes, enote_record.address_index.bytes, sizeof(enote_record.address_index));
```

Comment:
That `memcpy` looks suspicious to me, and I wonder why such a blunt "hammer" should be necessary to just copy something, but I don't know the codebase enough to really judge. Maybe just have a good look again?